File: app/src/main/java/com/tutv/android/domain/Series.java
package com.tutv.android.domain;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;
import androidx.room.Relation;

import java.util.List;

@Entity(tableName = "series")
public class Series {

    @PrimaryKey
    @ColumnInfo(name = "series_id")
    private int id;

    private int followers;

    private String name;
    private String seriesDescription;

    public String getBannerUrl() {
        return bannerUrl;
    }

    private String bannerUrl;

    public String getPosterUrl() {
        return posterUrl;
    }

    private String posterUrl;

    @Ignore List<Season> seasons;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSeriesDescription() {
        return seriesDescription;
    }

    public void setSeriesDescription(String seriesDescription) {
        this.seriesDescription = seriesDescription;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getFollowers() {
        return followers;
    }

    public void setFollowers(int followers) {
        this.followers = followers;
    }

    public List<Season> getSeasons() {
        return seasons;
    }

    public void setSeasons(List<Season> seasons) {
        this.seasons = seasons;
    }

    public void setBannerUrl(String bannerUrl) {
        this.bannerUrl = bannerUrl;
    }

    public void setPosterUrl(String posterUrl) {
        this.posterUrl = posterUrl;
    }
}

File: app/src/main/java/com/tutv/android/domain/Series.kt
package com.tutv.android.domain

import androidx.room.PrimaryKey
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import com.tutv.android.domain.Season

@Entity(tableName = "series")
class Series {
    @PrimaryKey
    @ColumnInfo(name = "series_id")
    var id = 0
    var followers = 0
    var name: String? = null
    var seriesDescription: String? = null
    var bannerUrl: String? = null
    var posterUrl: String? = null


    @Ignore
    var seasons: List<Season>? = null
    @Ignore
    var reviews: List<Review>? = null
    var loggedInUserFollows: Boolean? = null
}